src/xlsx/mod.rs

```diff
@@ -2199,6 +2435,474 @@ pub(crate) fn path_to_zip_path<RS: Read + Seek>(zip: &ZipArchive<RS>, path: &str
 }
 
+pub mod pivot_cache {
+    use super::XlReader;
+    use crate::{CellErrorType, Data, XlsxError};
+    use quick_xml::events::attributes::Attribute;
+    use quick_xml::events::BytesStart;
+    use quick_xml::events::Event;
+    use quick_xml::name::QName;
+    use quick_xml::Decoder;
+    use std::collections::HashMap;
+    use std::io::{Read, Seek};
+
+    pub type Tag = Box<[u8]>;
+    pub type Value = Option<Box<[u8]>>;
+
+    // Get the target location of the pivot table's pivot cache definitions.
+    pub fn find_pivot_cache_definitions_from_pivot<RS>(
+        zip: &mut zip::ZipArchive<RS>,
+        path: &str,
+    ) -> Result<String, XlsxError>
+    where
+        RS: Read + Seek,
+    {
+        let (base_folder, file_name) = path.rsplit_once('/').expect("should be in a folder");
+        let rel_path = format!("{base_folder}/_rels/{file_name}.rels");
+        let mut xml = match super::xml_reader(zip, &rel_path) {
+            None => return Err(XlsxError::FileNotFound(rel_path.to_owned())),
+            Some(x) => x?,
+        };
+        let mut paths = vec![];
+        let mut buf = Vec::with_capacity(64);
+        loop {
+            buf.clear();
+            match xml.read_event_into(&mut buf) {
+                Ok(Event::Start(ref e)) if e.local_name().as_ref() == b"Relationship" => {
+                    let mut target = String::new();
+                    let mut is_pivot_cache_definitions_type = false;
+                    for a in e.attributes() {
+                        match a.map_err(XlsxError::XmlAttr)? {
+                            Attribute {
+                                key: QName(b"Target"),
+                                value: v,
+                            } => target = xml.decoder().decode(&v)?.into_owned(),
+                            Attribute {
+                                key: QName(b"Type"),
+                                value: v,
+                            } => is_pivot_cache_definitions_type = *v == b"http://schemas.openxmlformats.org/officeDocument/2006/relationships/pivotCacheDefinition"[..],
+                            _ => (),
+                        }
+                    }
+                    if is_pivot_cache_definitions_type {
+                        if let Some(target) = target.strip_prefix("../") {
+                            // this is an incomplete implementation, but should be good enough for excel
+                            let (parent, _) = base_folder
+                                .rsplit_once('/')
+                                .expect("Must be a parent folder");
+                            paths.push(format!("{parent}/{target}"));
+                        } else if target.is_empty() { // do nothing
+                        } else {
+                            paths.push(target);
+                        }
+                    }
+                }
+                Ok(Event::End(ref e)) if e.local_name().as_ref() == b"Relationships" => break,
```

Comment:
Ref can be removed, as previously mentioned
```suggestion
                Ok(Event::End(e)) if e.local_name().as_ref() == b"Relationships" => break,
```